pymatgen/io/vasp/outputs.py

```diff
@@ -874,6 +874,35 @@ def eigenvalue_band_properties(self):
                         cbm_kpoint = k
         return max(cbm - vbm, 0), cbm, vbm, vbm_kpoint == cbm_kpoint
 
+    def smart_efermi(self):
+        """
+        Check whether the Fermi level reported by VASP crosses a band. If it does,
+        and if the bandgap is nonzero, place the Fermi level in the middle of the
+        bandgap.
+        """
+        # finding the Fermi level is quite painful, as VASP can sometimes put it slightly
+        # inside a band
+        fermi_crosses_band = False
+        for spin_eigenvalues in eigenvalues.values():
+            eigs_below = np.any(spin_eigenvalues < self.efermi, axis=1)
+            eigs_above = np.any(spin_eigenvalues > self.efermi, axis=1)
+            if np.any(eigs_above & eigs_below):
+                fermi_crosses_band = True
```

Comment:
Also, `spin_eigenvalues` isn't defined at this point.  You can change this section to
```suggestion
        for spin_eigenvalues in self.eigenvalues.values():
            # drop weights and set shape nbands, nkpoints
            spin_eigenvalues = spin_eigenvalues[:, :, 0].transpose(1, 0)
            eigs_below = np.any(spin_eigenvalues < self.efermi, axis=1)
            eigs_above = np.any(spin_eigenvalues > self.efermi, axis=1)
            if np.any(eigs_above & eigs_below):
                fermi_crosses_band = True
```